src/states/node.rs

```diff
@@ -2482,20 +2482,32 @@ impl Node {
                 debug!("{:?} Disconnecting from timed out peer {:?}", self, peer_id);
                 let _ = self.crust_service.disconnect(peer_id);
             }
-            let transition = self.purge_invalid_rt_entries(outbox);
+
+            let transition = if cfg!(feature = "use-mock-crust") {
+                Transition::Stay
+            } else {
+                self.purge_invalid_rt_entries(outbox)
```

Comment:
But isn't any purge panic considered a bug, regardless of whether it happens during connection or while there are still messages in flight?